### PR TITLE
add a connect timeout to grpc_connector

### PR DESCRIPTION
**Pull request by ilrudie**
related #752

Add a connect timeout for grpc connector to prevent long reconnect delays

### CODE REVIEW

File: src/tls/boring.rs [hunk 1]
```diff
@@ -309,6 +309,7 @@ pub fn grpc_connector(
     http.set_keepalive(Some(Duration::from_secs(300)));
     http.set_keepalive_interval(Some(Duration::from_secs(75)));
     http.set_keepalive_retries(Some(9));
+    http.set_connect_timeout(Some(Duration::from_millis(100)));
```
Comment (howardjohn) [inline src/tls/boring.rs:312]: we probably want 10s, not 100ms. 100ms is actually pretty quick for a connection, especially cross-zone

**Comment by hzxuzhonghu** [comment]
make it same as sidecar xds poroxy

File: src/tls/boring.rs (hunk 1 above)
Comment (ilrudie) [reply src/tls/boring.rs:312]: per @hzxuzhonghu's comment, it looks like we may use the default from the envoy `go-control-plane` which is `5s`. I'm not supremely confident in that but don't see where we set a connect timeout. Curious to be proven wrong if I missed it though. How do we feel about `5s`?

Comment (howardjohn) [reply src/tls/boring.rs:312]: Its not from that library but still 5s: https://github.com/istio/istio/blob/4482b870e91ce66b6cc776700cb4c94f1edc8dca/pkg/istio-agent/xds_proxy.go#L334. 5s is good.

**Review by howardjohn — APPROVED** [review]

**Pull request merged**